controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -259,6 +268,41 @@ func (r *BareMetalHostReconciler) Reconcile(request ctrl.Request) (result ctrl.R
 	return
 }
 
+// Consume inspect.metal3.io/hardwaredetails when either
+// inspect.metal3.io=disabled or there are no existing HardwareDetails
+func (r *BareMetalHostReconciler) updateHardwareDetails(request ctrl.Request, host *metal3v1alpha1.BareMetalHost) (bool, error) {
+	reqLogger := r.Log.WithValues("baremetalhost", request.NamespacedName)
+	updated := false
+	if host.Status.HardwareDetails == nil || inspectionDisabled(host) {
```

Comment:
Should this condition be an && instead of an ||? Also, we might be able to not delete the annotation later (after consuming it) if this condition is an &&.

Reply:
> Should this condition be an && instead of an ||? That way, when the annotation is processed, then we won't read the HW details again, even if the annotation is not deleted.

No, in the [design doc](https://github.com/metal3-io/metal3-docs/pull/155/files) we agreed the following behavior wrt when to update the status:

- At any time when inspect.metal3.io=disabled
- When there is no existing HardwareDetails data in the Status

So in the logic above, we'll update the status if HardwareDetails aren't yet set, or always in the case where `inspectionDisabled(host)` is true

The annotation is immediately deleted, so we won't re-read the same annotation, but it means a user could potentially update the data in the case where inspection is disabled (which IMO makes sense, because in that case the data is externally managed and so we shouldn't prevent updates)

Reply:
> I think I would reverse this logic and return early if none of the conditions for consuming the annotation are met.

@dhellmann The reason I did it this way is the following code to remove the annotation is currently (intentionally) hit unconditionally, so we either remove it after consuming the annotation, or after it's ignored (as you mention in the other thread maybe this second case should generate an event to make it more visible?)

Reply:
OK, that makes sense. It wasn't clear why we always wanted to remove it, but I think if we can signal via an event that we're ignoring the input then it makes sense to remove it.

Reply:
Ok the latest version generates two events, one when consuming the annotation

```
Normal  UpdateHardwareDetails  10m   metal3-baremetal-controller  Setting HardwareDetails from annotation
```
And one when ignoring it:

```
Normal  RemoveAnnotation  11m   metal3-baremetal-controller  HardwareDetails annotation ignored, status already set and inspection is not disabled
```